Brian/recipe output

BrianStas opened the pull request:
Recipe output mostly completed by design. Not the best, could be edited to add different info but it's a start and this is a good model for us to all share so we can link backend and update as we need.
To test:
1. npm install
2. data is already in there for sample. Go to:
localhost3000/recipes/results
3. click button at the bottom.
4. scroll on items to confirm
5. click X at top right to return.

BrianStas committed:
added in sample data with basic bones for output page

Added in sample json data for testing purposes along with the initial page used for output. Currently that page is just a scrape of my design so will need to edit to align with what we actually have from API calls.

BrianStas committed:
updated formatting

fixed format to fit in the dashboard and be included there. Need to still edit in order for it to display correctly.

BrianStas committed:
updated style for nutrients

Updated the list output so that it would display the macro-nutrient numbers in an easily digestible format so that users can make an educated decision on their recipe choices

BrianStas committed:
mild design updates

just a bit of design updates to clear up the images and give the cards a better flow.

BrianStas committed:
style update

Trying out a new style to see if this might be better. Will get feedback and update

BrianStas committed:
adding motion and first details extension

Adding in a motion package to help with modern feel. Also added a basis for the recipe details page, though it still needs editing before it will feel right.

BrianStas committed:
slight update to handling

Updated handling the click-through on Let's Cook! button, but will need to figure out changes once we decide on data structure being pulled upon search

BrianStas committed:
refactor for back end data

refactored everything to match up with what I understand the back end will now provide. Moved folders to provide space for the input portion of recipe seeking

BrianStas committed:
fixed definition

Fixed recipe definition so it wouldn't cause issue with typescript.

BrianStas committed:
all pages now working

cleaned up the details page to correctly display without issue now. Going into testing to nail down the design

BrianStas committed:
another refactor

added in animation and therefore redid the whole format once more to fix it in.  Looks relatively good now, though only using a single sample. Will keep working til it looks how I want.

BrianStas committed:
Merge branch 'main' of https://github.com/chingu-voyages/v53-tier3-team-37 into brian/recipe-output

BrianStas committed:
merged main, edited back formatting

Mostly got this done, merged main in to cut down on merge conflicts. Just small issues, easy to resolve.

BrianStas committed:
deleted excess data

removed data files not being used to clear up code a bit

tigeryash reviewed (approved):
You need to use 'use client' at the top of you results/page.tsx. It's a client component since you're using hooks. Instead of making the ingredients and steps scrollable make the whole card component scroll. After check marking an ingredient reduce the opacity of the text and strikethrough (tailwind line-through) the text. There's no code that really shows how you're planning on handling more than one piece of data. You should color code the cards you made to differentiate the macros. Either to the text or background. There needs to be a button which will let the user add it to there food log. You should utilize more of the request from spoonacular. A description of the food and using the image would be good additions. Beside the ingredients and steps headers. You could add the cookingMinutes and the preparationMinutes. If you're planning on adding a slider to swipe through the recipes suggested. It would be bad UX to still be able to swipe while viewing the current recipe. Make the detailed view of the recipe full screen. 

After I added the 'use client' at the top of ResultsPage the code works.

BrianStas committed:
updated output

Made requested changes for the most part. Description is now on front of card in scrollable box, checkboxes mark off ingredients on back and the macro boxes are colored.

skno27 commented:
Update sounds good. I trust you guys to figure out any remaining issue later on

skno27 reviewed (approved):
sounds like yash's concerns were addressed. approving
